libcnb-cargo/src/package/error.rs

```diff
@@ -228,3 +228,22 @@ impl From<RewriteBuildpackageRelativePathDependenciesToAbsoluteError> for Error
         }
     }
 }
+
+impl From<FindCargoWorkspaceError> for Error {
+    fn from(value: FindCargoWorkspaceError) -> Self {
+        match value {
+            FindCargoWorkspaceError::GetCargoEnv(error) => Error::GetCargoBin(error),
+            FindCargoWorkspaceError::SpawnCommand(error) => Error::SpawnCargoCommand(error),
+            FindCargoWorkspaceError::CommandFailure(exit_status) => {
+                Error::CargoCommandFailure(exit_status_or_unknown(exit_status))
+            }
+            FindCargoWorkspaceError::GetParentDirectory(path) => Error::GetWorkspaceDirectory(path),
+        }
+    }
+}
+
+fn exit_status_or_unknown(exit_status: ExitStatus) -> String {
```

Comment:
I see no reason why this function should take ownership of `ExitCode`. Not a huge deal, but actually prevents us to move the error display logic into the `Display` implementation for `Error` (via `thiserror`). See comment there.